### PR TITLE
[GITHUB APPS] release branch

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description
Feature branch for docs to support GitHub apps release

* New integration guide with CircleCI GitHub App
* * Updated Quickstart guide (getting-started) and Sign up and try doc (first-steps) to incorporate GH Apps accounts
* Create project tutorial now shows both **Set up project** and **Create project** options
* Banners to direct readers on how to determine whether they integrated through GitHub OAuth or the GitHub app
* New doc to walk through invite team process
* **Roles and permissions overview** and **Manage roles and permissions** guides now under **Security and permissions**
* Banners to show SSH rerun not supported
* Banner to show config editor not supported for standalone outside project create flow
* Banners to direct readers to information on using the correct project slug in API calls based on integration type

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/api-developers-guide.md [hunk 1]
```diff
@@ -102,16 +110,24 @@ The `project_slug` is included in the payload when pulling information about a p
...
 {: #gitlab-saas-support-projects }
 
-For GitLab Saas Support, organization as well as project names do not serve as identifiers, and are not part of project slugs. GitLab projects currently use a new slug format:
+**GitHub authentication with CircleCI is changing**. Starting August 2023 when you authenticate your CircleCI account with GitHub, you may find this will be done through our GitHub App, rather than the GitHub OAuth app. You can see which account type you have by heading to the CircleCI web app and inspecting the URL in your browser:
+<br>
+* This style of URL indicates you authenticated with the **GitHub App**: `https://app.circleci.com/pipelines/circleci/UTxCZDiJ9MLGLC8hR1ZDmg`
+* This style of URL indicates you authenticated with the **GitHub OAuth app**: `https://app.circleci.com/github/<your GitHub username>`
+<br>
+If you authenticated with **GitHub App**, this section is relevant to you. For **GitHub OAuth**, refer to the [above section](#github-and-bitbucket-projects).
+{: class="alert alert-info"}
```
Comment (kbatuigas) [inline jekyll/_cci2/api-developers-guide.md:124]: Nitpick: Still need to troubleshoot why banner isn't rendering

File: jekyll/_cci2/api-developers-guide.md [hunk 2]
```diff
@@ -85,10 +85,18 @@ v1) return EDN if no accept header is specified.
...
+**GitHub authentication with CircleCI is changing**. Starting August 2023 when you authenticate your CircleCI account with GitHub, you may find this will be done through our GitHub App, rather than the GitHub OAuth app. You can see which account type you have by heading to the CircleCI web app and inspecting the URL in your browser:
+<br>
+* This style of URL indicates you authenticated with the **GitHub App**: `https://app.circleci.com/pipelines/circleci/UTxCZDiJ9MLGLC8hR1ZDmg`
+* This style of URL indicates you authenticated with the **GitHub OAuth app**: `https://app.circleci.com/github/<your GitHub username>`
+<br>
+If you authenticated with the **GitHub OAuth app**, this section is relevant to you. For **GitHub App**, refer to the [following section](#gitlab-saas-support-projects).
+{: class="alert alert-info"}
```
Comment (kbatuigas) [inline jekyll/_cci2/api-developers-guide.md:97]: Nitpick: Still need to troubleshoot why banner isn't rendering

File: jekyll/_data/sidenav.yml [hunk 3]
```diff
@@ -30,8 +32,10 @@ en:
             link: config-editor
       - name: VCS integration
         children:
-          - name: GitHub integration overview
+          - name: GitHub OAuth integration overview
             link: github-integration
+          - name: GitHub apps integration overview
```
Comment (rosieyohannan) [inline jekyll/_data/sidenav.yml:37]:
```suggestion
          - name: GitHub Apps integration overview
```

Comment (kbatuigas) [reply jekyll/_data/sidenav.yml]: @rosieyohannan I went through the PR and changed a lot of the "GitHub Apps" instances to "GitHub App" or "the CircleCI GitHub App"... shall we do that for the sidenav as well?

**Review by kbatuigas — APPROVED** [review]

**Pull request merged**